aws_jobs/cyhy-data-extract.py

```diff
@@ -136,30 +135,15 @@ def query_data(collection, query, tbz_file, tbz_filename, end_of_data_collection
     with open(json_filename, "w") as collection_file:
         collection_file.write("[")
 
-        last_id = None
-        while True:
-            if last_id is None:
-                result = collection.find(
-                    query, {"key": False}, sort=[("_id", pymongo.ASCENDING)]
-                ).limit(PAGE_SIZE)
```

Comment:
@KyleEvers and @mcdonnnj, why was the `find` query was paginated before?  Is there a memory constraint or something that causes a failure in AWS if we _don't_ paginate?

Reply:
The previous method converted all documents retrieved into a JSON string at once. This had a very large memory overhead and certain queries would consume enough memory in this process to crash the AWS instance being used before pagination. We are now retrieving and processing a single document at a time and the memory overhead is drastically lower. I profiled the memory consumption locally and then tested against production to verify no runtime issues.

Reply:
Excellent.  Do you mind adding a comment in the code that briefly describes this decision and the subsequent improvement?  What you wrote is fine, and you can leave out the last sentence.

Reply:
Description added by @mcdonnnj in 95aa472.